Javascript/javascript1/week4

JyothiNandyala opened the pull request:
Adding Javascript1 week4 home work file.

ajgcph2019 commented on javascript/javascript1/week4/voiceassistent.js:
```diff
@@ -0,0 +1,104 @@
+        return "please enter the command in specific formate";
+    }
+    
+    else if(commandArray[0]==="hello"){
```
There are string methods like 'startsWith' and 'endsWith' .Informing for your reference
` const commandArray=command.split(' ');

  if (commandArray.startsWith('hello my name is')) {
    name= commandArray[commandArray.length-1];
    return `Nice to meet you ${name}`;
  }`

JyothiNandyala replied:
yes,i made the changes

ajgcph2019 commented on javascript/javascript1/week4/voiceassistent.js:
```diff
@@ -0,0 +1,104 @@
+    return getMathFunction(commandArray);
+}
+}
+//Mathamatics it will calculate the mathematics operation and return the result.
```
Do you use Prettier for fomatting? There are extra spaces and few issues with alignment of code

JyothiNandyala replied:
i have changed the code now.

ajgcph2019 commented on javascript/javascript1/week4/voiceassistent.js:
```diff
@@ -0,0 +1,104 @@
+    let  operation=commandArray[3];
+    let a=parseInt(commandArray[2]);
+    let b=parseInt(commandArray[4]);
+    
```
Use meaningful variable names instead of a and b

JyothiNandyala replied:
yes i did it.thank you for remembering.

JyothiNandyala commented:
Hi Anna,

I did the changes can i merge my pull request,could you please let me know.

thank you.

On Fri, Nov 27, 2020 at 9:22 PM Anna John George <notifications@github.com>
wrote:

> *@ajgcph2019* requested changes on this pull request.
> ------------------------------
>
> In javascript/javascript1/week4/voiceassistent.js
> <https://github.com/JyothiNandyala/hyf-homework/pull/6#discussion_r531669763>
> :
>
> > @@ -0,0 +1,104 @@
> +"use strict"
> +
> +//voice Assistant
> +let name='';
> +let todos=[];
> +
> +function getReply(command){
> +    const commandArray=command.split(' ');
> +
> +
> +    if(typeof command!=="string"){
> +        return "please enter the command in specific formate";
> +    }
> +
> +    else if(commandArray[0]==="hello"){
>
> There are string methods like 'startsWith' and 'endsWith' .Informing for
> your reference
> ` const commandArray=command.split(' ');
>
> if (commandArray.startsWith('hello my name is')) {
> name= commandArray[commandArray.length-1];
> return Nice to meet you ${name};
> }`
> ------------------------------
>
> In javascript/javascript1/week4/voiceassistent.js
> <https://github.com/JyothiNandyala/hyf-homework/pull/6#discussion_r531671977>
> :
>
> > +    const minutes = commandArray[4];
> +    const milliSeconds = minutes*60*1000;
> +
> +
> +    function setTimer(){
> +        console.log("Timer done");
> +}
> +    console.log("Timer set for "+commandArray[4]+ " " +commandArray[5]);
> +    setTimeout(setTimer, milliSeconds);
> +    return "Timer is set so just wait";
> +}
> +if(command){
> +    return getMathFunction(commandArray);
> +}
> +}
> +//Mathamatics it will calculate the mathematics operation and return the result.
>
> Do you use Prettier for fomatting? There are extra spaces and few issues
> with alignment of code
> ------------------------------
>
> In javascript/javascript1/week4/voiceassistent.js
> <https://github.com/JyothiNandyala/hyf-homework/pull/6#discussion_r531673068>
> :
>
> > +}
> +    console.log("Timer set for "+commandArray[4]+ " " +commandArray[5]);
> +    setTimeout(setTimer, milliSeconds);
> +    return "Timer is set so just wait";
> +}
> +if(command){
> +    return getMathFunction(commandArray);
> +}
> +}
> +//Mathamatics it will calculate the mathematics operation and return the result.
> +
> +function getMathFunction(commandArray){
> +    let  operation=commandArray[3];
> +    let a=parseInt(commandArray[2]);
> +    let b=parseInt(commandArray[4]);
> +
>
> Use meaningful variable names instead of a and b
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/JyothiNandyala/hyf-homework/pull/6#pullrequestreview-540066358>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARCQJSDJ2YXBBMRELD5B5GDSR7DKJANCNFSM4TZ6M7AQ>
> .
>

JyothiNandyala commented:
made the changes based on mentors feedback.
